### ISSUE TITLE
Return correct Content-Type for JSON-LD resources (from file extension)

### ISSUE THREAD

**Issue by csarven** [solid/node-solid-server#413]
If Content-Type is derived from the file extension, return the correct mime-type in the response headers. Appears to default to text/turtle.

e.g.,
- `foo.jsonld` returns text/turtle Content-Type
- `foo.json` returns application/json with HTTP 501 (this is probably a separate issue)

Possibly related:
- `foo` (in JSON-LD) returns text/turtle Content-Type
- `"Accept: application/ld+json"` returns `Error translating between RDF formats` (trying to parse it as Turtle before trying serialise as JSON-LD - see also https://github.com/solid/node-solid-server/issues/226)

**Comment by dmitrizagidulin** [comment]
I suspect that whatever library we're using to derive content type from file extensions doesn't recognize `.jsonld`.. We should look into it (and if there's nothing to be done there, handle it separately in the code).

**Comment by dmitrizagidulin** [comment]
Let's tackle these issues one at a time.

> `foo.jsonld` should return `application/ld+json` as its Content-Type

The main culprit is that the [mime](https://www.npmjs.com/package/mime) npm package that `node-solid-server` is using to translate from a file extension to a MIME Type just doesn't recognize `application/ld+json` as a valid type. Try this out in your terminal:

```bash
npm install mime
node
> var mime = require('mime')
> mime.extension('application/json')
'json'
> mime.extension('application/ld+json')
undefined
```

Same thing in the other direction:

```bash
> mime.lookup('.json')
'application/json'
> mime.lookup('.jsonld')
'application/octet-stream'
```

I suspect this is because the JSON-LD MIME Type is still undergoing the IANA standardization process? I think a temporary fix for this might be adding in a hardcoded custom check to `node-solid-server`, to test specifically for the `.jsonld` extension.

**Comment by csarven** [comment]
https://www.iana.org/assignments/media-types/application/ld+json is standardised as it gets?

**Comment by dmitrizagidulin** [comment]
@csarven aha! Excellent. Let's file an issue on the `mime` npm package, to add it?

**Comment by melvincarvalho** [comment]
FYI: A few sources:

http://www.iana.org/assignments/media-types/media-types.xhtml
http://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types
http://hg.nginx.org/nginx/raw-file/default/conf/mime.types

These are aggregated here:

https://github.com/jshttp/mime-db

Github uses this, for example.

**Comment by dmitrizagidulin** [comment]
Can somebody open an issue on mime-db, or make a PR for jsonld?

**Comment by melvincarvalho** [comment]
JSON-LD is supported by mime-db, isnt it?  What would the pull request be for?

**Comment by dmitrizagidulin** [comment]
@melvincarvalho oh, whoops, yeah. I was hoping the `mime` npm package drew from that, but looks like it doesnt.

**Comment by dmitrizagidulin** [comment]
This has been fixed in PR #430. Closing.
